### PR TITLE
bug: Silence remaining nuisance alarms

### LINKED ISSUE #1123: Capture exceptions that are connection loss related
Various connection breakage throws exceptions into Sentry. These are not bugs we plan on fixing, so they should be captured and dismissed as connection breaks.

Sentry bug(s):
- https://sentry.prod.mozaws.net/operations/autopush-prod/issues/1673127/
- https://sentry.prod.mozaws.net/operations/autopush-prod/issues/643290/
- https://sentry.prod.mozaws.net/operations/autopush-prod/issues/1737236/

And a variety of Connect timeout exceptions, that may need to be handled differently:
- https://sentry.prod.mozaws.net/operations/autopush-prod/issues/1814177/
- https://sentry.prod.mozaws.net/operations/autopush-prod/issues/1814153/
- etc.

### PR DESCRIPTION
**Pull request by jrconlin**
Closes #1123

### CODE REVIEW

File: autopush/router/apnsrouter.py [hunk 1]
```diff
@@ -157,7 +158,7 @@ def _route(self, notification, router_data):
                                        self._base_tags,
                                        application=rel_channel,
                                        reason="connection_error"))
-        except HTTP20Error:
+        except (HTTP20Error, socket.error):
```
Comment (pjenvey) [inline autopush/router/apnsrouter.py]: This makes the SSLError block below dead code (SSLError derives from socket.error). Go ahead and kill it

File: autopush/main.py [hunk 2]
```diff
@@ -105,7 +106,10 @@ def run(self):
         """Start the services and run the reactor"""
         reactor.suggestThreadPoolSize(constants.THREAD_POOL_SIZE)
         self.startService()
-        reactor.run()
+        try:
+            reactor.run()
+        except ConnectionDone:
```
Comment (pjenvey) [inline autopush/main.py]: this one's not right but I can see what lead you towards it: our stack trace order in sentry is screwed up (logged #1134). Look at it again in reverse =]

File: autopush/web/base.py [hunk 3]
```diff
@@ -238,7 +238,10 @@ def _response_err(self, fail):
         running.
 
         """
+        from twisted.internet.error import ConnectionDone
         fmt = fail.value.message or 'Exception'
+        if fail.value.__class__ in [ConnectionDone]:
```
Comment (pjenvey) [inline autopush/web/base.py]: why not isinstance?

File: autopush/main.py (hunk 2 above)
Comment (jrconlin) [reply autopush/main.py]: I'm not so sure. I was looking at https://sentry.prod.mozaws.net/operations/autopush-prod/issues/643290/. That seems to indicate that the `ConnectionDone` error is happening as part of `reactor.run()` here.

Comment (pjenvey) [reply autopush/main.py]: that one's due to web.base _response_err logging it, you can see it in the next to last stack frame

File: autopush/web/base.py (hunk 3 above)
Comment (jrconlin) [reply autopush/web/base.py]: wasn't sure if there were going to be others.

Comment (pjenvey) [reply autopush/web/base.py]: commonly unknown but so nice: isinstance arg 2 can be a tuple of types

File: autopush/main.py (hunk 2 above)
Comment (jrconlin) [reply autopush/main.py]: I have no idea how the hell this got reverted. Removing this try since the ConnectionDone is handled at a lower layer.

**Pull request merged**